usecases/config/environment.go

```diff
@@ -631,8 +631,13 @@ func parseResourceUsageEnvVars() (ResourceUsage, error) {
 func parseClusterConfig() (cluster.Config, error) {
 	cfg := cluster.Config{}
 
-	if v := os.Getenv("CLUSTER_HOSTNAME"); v != "" {
-		cfg.Hostname = v
+	// by default memberlist assigns hostname to os.Hostname() incase hostname is empty
+	// ref: https://github.com/hashicorp/memberlist/blob/3f82dc10a89f82efe300228752f7077d0d9f87e4/config.go#L303
+	// it's handled at parseClusterConfig step to be consistent from the config start point and conveyed to all
+	// underlying functions see parseRAFTConfig(..) for example
+	cfg.Hostname = os.Getenv("CLUSTER_HOSTNAME")
+	if cfg.Hostname == "" {
+		cfg.Hostname, _ = os.Hostname()
```

Comment:
thoughts(non-blocking): Do we want to handle here the fact that we might have an error ? Not sure what we would do just raising the question

Reply:
yeah i had the same thought, but not sure if it's really an error, what i could do that we log warn that we will use the default od hostname